client/boot/common.js

```diff
@@ -36,41 +36,34 @@ const switchUserLocale = ( currentUser, reduxStore ) => {
 
 const setupContextMiddleware = reduxStore => {
 	page( '*', ( context, next ) => {
-		const parsed = url.parse( location.href, true );
-
-		// Decode the pathname by default (now disabled in page.js)
-		context.pathname = decodeURIComponent( context.pathname );
-
-		context.store = reduxStore;
-
-		// Break routing and do full load for logout link in /me
-		if ( context.pathname === '/wp-login.php' ) {
-			window.location.href = context.path;
-			return;
-		}
-
-		// set `context.query`
-		const querystringStart = context.canonicalPath.indexOf( '?' );
-
-		if ( querystringStart !== -1 ) {
-			context.query = qs.parse( context.canonicalPath.substring( querystringStart + 1 ) );
-		} else {
-			context.query = {};
-		}
-
+		// page.js url parsing is broken so we had to disable it with `decodeURLComponents: false`
```

Comment:
What exactly is broken?

Reply:
query parameters parsing is broken. For instance if you have `/log-in?redirect_to=https%3A%2F%2Fwordpress.com%2Fthemes%3Fretry%3D1%26test%3D2` context.query is going to be
```json
{
"redirect_to": "https://wordpress.com/themes?retry=1"
"test": "2"
}
```
It's a [known issue](https://github.com/visionmedia/page.js/issues/216) in page.js, this is why they created this `decodeURLComponents: false` option.

Reply:
Thanks for explaining! Would be good to include a link to the issue in the code comment IMO :slightly_smiling_face:

Reply:
Some more related discussion at https://github.com/Automattic/wp-calypso/pull/19418#issuecomment-341390044